Src/Library/Binder/RequestBinder.cs

```diff
@@ -149,7 +155,29 @@ private static void BindRouteValues(TRequest req, RouteValueDictionary routeValu
     private static void BindQueryParams(TRequest req, IQueryCollection query, List<ValidationFailure> failures)
     {
```

Comment:
this way of repeatedly iterating properties in a nested manner doesn't look that performant to me. since it's gonna happen for every single request, each time. most likely why binding to complex objects from query params like this is not supported in minimal apis. since it's not supported by minimal apis, we don't really need to support it anyway if we can't do it efficiently. also the allocation of dictionaries and json strings and repeatedly calling the json serializer is also not that elegant and will add to the garbage collector pressure.

so going forward, if someone really must do it, they'd either have to add a static `TryParse()` method to the the type or implement a [custom binder](https://fast-endpoints.com/docs/model-binding#diy-request-binding) themselves.

Reply:
Total number of iterations will be the same as for simple query binding.
The main problem here is dictionary parsing. I'll see how we can avoid it. Also I'll try to run some performance test to compare/